.github/workflows/CI-build-test.yml

```diff
@@ -1,69 +1,65 @@
-name: CI-build-test
+name: Build-and-Test
 
-run-name: "Workflow performing CI steps: build, unit testing, integration testing"
+run-name: "Workflow CI/CD Steps: build, unit and integration testing"
 
 on:
   workflow_call:
 
 env:
-  RUST_BACKTRACE: 1
+  IMAGE_NAME: zencash/sc-ci-base
+  IMAGE_TAG: noble_rust-stable_latest
+  DOCKER_COMPOSE_CMD: "docker compose -f ${GITHUB_WORKSPACE}/ci/docker-compose.yml run --rm cargo-container"
+  DOCKER_BUILD_DIR: /build
+  DOCKER_CARGO_HOME: /tmp/.cargo
 
 jobs:
-  build-test:
-    strategy:
-      matrix:
-        os: [ubuntu-latest]
-        # os: [ubuntu-latest, windows-latest, macos-latest]
-    runs-on: ${{ matrix.os }}
+  build-and-test:
+    runs-on: ubuntu-latest
     steps:
-      - name: Checkout branch
+      - name: Git checkout
         uses: actions/checkout@v4
+
       - name: Set up cargo cache
         uses: actions/cache@v4
-        continue-on-error: false
+        if: ${{ !env.ACT }}
+        continue-on-error: true
         with:
           path: |
             ~/.cargo/bin/
             ~/.cargo/registry/index/
             ~/.cargo/registry/cache/
             ~/.cargo/git/db/
             target/
-          key: ${{ runner.os }}-build-test-cargo-${{ hashFiles('**/Cargo.lock') }}
-          restore-keys: ${{ runner.os }}-build-test-cargo-
```

Comment:
I think `restore-keys` should be reintroduced, it was useful as a fallback in case an exact match on `key` was not found.
Please also apply to other workflows.

Reply:
Done